### PR TITLE
docs: add CNI and host_network docs

### PR DESCRIPTION
**Pull request by nickethier**

### CODE REVIEW

File: command/agent/config.go [hunk 1]
```diff
@@ -277,6 +277,10 @@ type ClientConfig struct {
 	// specified colon delimited
 	CNIPath string `hcl:"cni_path"`
 
+	// CNIConfigDir is the directory where CNI network configuration is located. The
+	// client will use this path when fingerprinting CNI networks.
+	CNIConfigDir string `hcl:"cni_config_dir`
```
Comment (shoenig) [inline command/agent/config.go:282]:
```suggestion
	CNIConfigDir string `hcl:"cni_config_dir"`
```

File: website/pages/docs/configuration/client.mdx [hunk 2]
```diff
@@ -151,6 +154,9 @@ driver) but will be removed in a future release.
 - `host_volume` <code>([host_volume](#host_volume-stanza): nil)</code> - Exposes
   paths from the host as volumes that can be mounted into jobs.
 
+- `host_network` <code>([host_network](#host_network-stanza): nil)</code> - Registers
+  addtional host networks with the node that can be selected when port mapping.
```
Comment (shoenig) [inline website/pages/docs/configuration/client.mdx:158]:
```suggestion
  additional host networks with the node that can be selected when port mapping.
```

File: website/pages/docs/configuration/client.mdx [hunk 3]
```diff
@@ -372,6 +378,35 @@ client {
...
+```hcl
+client {
+  host_network "public" {
+    cidr = "203.0.113.0/24"
+    reserved_ports = "22,80"
+  }
+}
+```
+
+#### `host_network Parameters
```
Comment (shoenig) [inline website/pages/docs/configuration/client.mdx:398]:
```suggestion
#### `host_network` Parameters
```

File: website/pages/docs/configuration/client.mdx [hunk 4]
```diff
@@ -372,6 +378,35 @@ client {
...
+
+- `cidr` `(string: "")` - Specifies a cidr block of addresses to match against.
+    If an address is found on the node that is contained by this cidr block, the
+    host network will be registered with it.
+
+- `interface` `(string: "")` - Filters searching of addresses to a specific interface
```
Comment (shoenig) [inline website/pages/docs/configuration/client.mdx:404]:
```suggestion
- `interface` `(string: "")` - Filters searching of addresses to a specific interface.
```

File: website/pages/docs/configuration/client.mdx [hunk 5]
```diff
@@ -372,6 +378,35 @@ client {
...
+
+- `reserved_ports` `(string: "")` - Specifies a comma-separated list of ports to
+  reserve on all fingerprinted network devices. Ranges can be specified by using
+  a hyphen separated the two inclusive ends.
```
Comment (shoenig) [inline website/pages/docs/configuration/client.mdx:408]:
```suggestion
  a hyphen separating the two inclusive ends.
```

File: website/pages/docs/job-specification/network.mdx [hunk 6]
```diff
@@ -224,6 +231,73 @@ network {
...
+The Nomad client will build the correct [capabilities arguments](https://github.com/containernetworking/cni/blob/v0.8.0/CONVENTIONS.md#well-known-capabilities) for the portmap plugin based on the defined port stanzas.
+
+### Host Networks
+
+If `host_network` is set for a port, Nomad will schedule the allocations on a node which has defined a `host_network` with the given name.
+If not set the "default" host network is used which is commonly the address with a default route associated with it.
+
+
+```hcl
+network {
```
Comment (shoenig) [inline website/pages/docs/job-specification/network.mdx:285]: probably needs to be hclfmt'd 

```hcl
network {
  mode = "bridge"
  # define a port to use for public https traffic
  port "https" {
    static       = 443
    to           = 8080
    host_network = "public"
  }
  # define a port that is only exposed to private traffic
  port "admin" {
    to           = 9000
    host_network = "private"
  }
}
```

File: website/pages/docs/configuration/client.mdx [hunk 7]
```diff
@@ -372,6 +378,35 @@ client {
 - `read_only` `(bool: false)` - Specifies whether the volume should only ever be
   allowed to be mounted `read_only`, or if it should be writeable.
 
+### `host_network` Stanza
+
+The `host_network` stanza is used to register additional host networks with
+the node that can be used when port mapping.
+
+The key of the stanza corresponds to the name of the network used in the
+[`host_network`](/docs/job-specification/network#host-network).
```
Comment (lgfa29) [inline website/pages/docs/configuration/client.mdx:387]: Sorry for the late review, but the link here is broken. It needs to be `/docs/job-specification/network#host-networks` (with an `s` at the end)

Comment (nickethier) [reply website/pages/docs/configuration/client.mdx:387]: Good catch I’ll fix it in my next round of docs

**Pull request merged**